edu-services/gradebook-service/api/src/java/org/sakaiproject/service/gradebook/shared/GradebookService.java

```diff
@@ -683,16 +683,15 @@ public String getAssignmentScoreString(String gradebookUid, Long assignmentId, S
 	
 	/**
 	 * Get student's assignment's score as string.
-	 * This is provided for backward compatibility only.
+	 * This method will attempt to lookup the Long value if provided as a String
+	 * You should use {@link #getAssignmentScoreString(String, Long, String)}
 	 * 
 	 * @param gradebookUid
 	 * @param assignmentName
 	 * @param studentUid
 	 * @return String of score
 	 * 
-	 * @deprecated See {@link #getAssignmentScoreString(String, Long, String)}
 	 */
-	@Deprecated
 	public String getAssignmentScoreString(String gradebookUid, String assignmentName, String studentUid)
```

Comment:
Yeah, maybe working in reverse (number looking fallback) could be better too. My biggest concern with that is in the long run, there would be more numeric ids than non-numeric in the database, so it would waste an extra select and rarely find anything. ID's become the default after this. So it's more of a do we care about eliminating that extra processing (that will easily be 99% of cases soon) or are more worried about numeric title matches. I was thinking of just putting in in as "500L" rather than "500" then that would have to eliminate 99.99% of the conflicts and only cause a problem if someone was trying to. (Which wouldn't matter after this anyway)